src/ParseFile.js

```diff
@@ -189,10 +212,19 @@ class ParseFile {
         this._previousSave = controller.saveFile(this._name, this._source, options).then((res) => {
           this._name = res.name;
           this._url = res.url;
+          this._data = null;
           return this;
         });
       } else if (this._source.format === 'uri') {
-        this._previousSave = controller.saveUri(this._name, this._source, options).then((res) => {
+        this._previousSave = controller.download(this._source.uri).then((result) => {
+          const newSource = {
+            format: 'base64',
+            base64: result.base64,
+            type: result.contentType,
+          };
+          this._data = result.base64;
+          return controller.saveBase64(this._name, newSource, options);
+        }).then((res) => {
```

Comment:
I know it isn't your code, but I'm surprised that this function doesn't always return a promise!  

if !this._previousSave, then this wont return a promise and anything that then's this func will error.

maybe in practive there is always a previous state when this is called???

Reply:
I don't know why thats there. Been there since the beginning of the SDK. But this always returns a Promise.

Reply:
Want me to clean it up or leave it?

Reply:
definitely your call

the class could definitely use some factoring out in any number of places (base64 comma location stuff for instance), but your changes look good to me and it works as is.